fetcher/util.go

```diff
@@ -122,14 +155,35 @@ func fetchFileConcurrently(req fetchRequest, concurrency int) (body []byte, err
 	}
 
 	var bytesBody []byte
-	if req.bzip2 {
+	switch req.mimeType {
+	case mimeTypeXML, mimeTypeTxt:
+		bytesBody = buf.Bytes()
+	case mimeTypeBzip2:
 		var b bytes.Buffer
 		if _, err := b.ReadFrom(bzip2.NewReader(bytes.NewReader(buf.Bytes()))); err != nil {
-			return body, err
+			return nil, xerrors.Errorf("Failed to open bzip2 file. err: %w", err)
+		}
+		bytesBody = b.Bytes()
+	case mimeTypeXz:
+		var b bytes.Buffer
+		r, err := xz.NewReader(bytes.NewReader(buf.Bytes()))
+		if err != nil {
+			return nil, xerrors.Errorf("Failed to open xz file. err: %w", err)
+		}
+		if _, err := b.ReadFrom(r); err != nil {
+			return nil, xerrors.Errorf("Failed to read xz file. err: %w", err)
+		}
+		bytesBody = b.Bytes()
+	case mimeTypeGzip:
+		var b bytes.Buffer
+		r, err := gzip.NewReader(bytes.NewReader(buf.Bytes()))
+		if err != nil {
+			return nil, xerrors.Errorf("Failed to open gzip file. err: %w", err)
+		}
+		if _, err := b.ReadFrom(r); err != nil {
+			return nil, xerrors.Errorf("Failed to read gzip file. err: %w", err)
 		}
 		bytesBody = b.Bytes()
-	} else {
-		bytesBody = buf.Bytes()
 	}
 
```

Comment:
Is it better to use `bytes.Buffer` instead of `[]byte` at first and set it to `[]byte` at the time of return?
The following code is an image.
```diff
diff --git a/fetcher/util.go b/fetcher/util.go
index 968c722..fd1fc11 100644
--- a/fetcher/util.go
+++ b/fetcher/util.go
@@ -154,18 +154,15 @@ func fetchFileConcurrently(req fetchRequest, concurrency int) (body []byte, err
 		return nil, xerrors.Errorf("Failed to write to output stream: %w", err)
 	}
 
-	var bytesBody []byte
+	var b bytes.Buffer
 	switch req.mimeType {
 	case mimeTypeXML, mimeTypeTxt:
-		bytesBody = buf.Bytes()
+		b = buf
 	case mimeTypeBzip2:
-		var b bytes.Buffer
 		if _, err := b.ReadFrom(bzip2.NewReader(bytes.NewReader(buf.Bytes()))); err != nil {
 			return nil, xerrors.Errorf("Failed to open bzip2 file. err: %w", err)
 		}
-		bytesBody = b.Bytes()
 	case mimeTypeXz:
-		var b bytes.Buffer
 		r, err := xz.NewReader(bytes.NewReader(buf.Bytes()))
 		if err != nil {
 			return nil, xerrors.Errorf("Failed to open xz file. err: %w", err)
@@ -173,9 +170,7 @@ func fetchFileConcurrently(req fetchRequest, concurrency int) (body []byte, err
 		if _, err := b.ReadFrom(r); err != nil {
 			return nil, xerrors.Errorf("Failed to read xz file. err: %w", err)
 		}
-		bytesBody = b.Bytes()
 	case mimeTypeGzip:
-		var b bytes.Buffer
 		r, err := gzip.NewReader(bytes.NewReader(buf.Bytes()))
 		if err != nil {
 			return nil, xerrors.Errorf("Failed to open gzip file. err: %w", err)
@@ -183,10 +178,8 @@ func fetchFileConcurrently(req fetchRequest, concurrency int) (body []byte, err
 		if _, err := b.ReadFrom(r); err != nil {
 			return nil, xerrors.Errorf("Failed to read gzip file. err: %w", err)
 		}
-		bytesBody = b.Bytes()
 	}
-
-	return bytesBody, nil
+	return b.Bytes(), nil
 }
 
 func fetchFileWithUA(req fetchRequest) (body []byte, err error) {

```